samples/sample_common/include/sample_utils.h

```diff
@@ -992,7 +992,30 @@ class CAutoTimer
 
 mfxStatus ConvertFrameRate(mfxF64 dFrameRate, mfxU32* pnFrameRateExtN, mfxU32* pnFrameRateExtD);
 mfxF64 CalculateFrameRate(mfxU32 nFrameRateExtN, mfxU32 nFrameRateExtD);
-mfxU16 GetFreeSurfaceIndex(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
+
+#ifndef MSDK_INVALID_SURF_IDX
+    #define MSDK_INVALID_SURF_IDX 0xFFFF
+#endif
+
+template <class T>
+mfxU16 GetFreeSurfaceIndex(T* pSurfacesPool, mfxU16 nPoolSize)
+{
+    if (pSurfacesPool)
+    {
+        for (mfxU16 i = 0; i < nPoolSize; i++)
```

Comment:
use ```std::find_if```

Reply:
i think we have more code in this case 
```
mfxU16 i = 0;
auto it = std::find_if(pSurfacesPool, pSurfacesPool + nPoolSize, [&i](const T & surface)
{
    i++;
    return (0 == surface.Data.Locked);
});

if (it != (pSurfacesPool + nPoolSize))
    return i-1;
```

Reply:
@mgonchar , please comment. We need to address the regression ASAP

Reply:
I'm ok to merge it as is.

I thought about different variant, but in general I agree with you
```
auto it = std::find_if(pSurfacesPool, pSurfacesPool + nPoolSize, [](const T & surface)
{    return 0 == surface.Data.Locked; });
if (it == pSurfacesPool + nPoolSize)
   return MSDK_INVALID_SURF_IDX;

return it - pSurfacesPool;
```